### DISCUSSION TITLE
Request: video walking through and explaining code-base

### DISCUSSION THREAD

**Discussion by rcyost** [winedarksea/AutoTS#204] [General]
It seems like to get other involved, it would be best for Colin to screen record and walk us through the code base. Writing up the documentation fully and completely would probably take too long. But there seems to be people that would like to contribute, but there isn't complete enough documentation to get others up to 100% speed. For example, I'd like to contribute by adding a feature pipeline element such as https://tsfresh.readthedocs.io/en/ but feel like I need more than the existing documentation. For example, one of the coolest parts is the genetic algorithm that searches the models and hyper-parameters, but is little talked about in the documentation.

**Comment by winedarksea** [comment]
I appreciate the enthusiasm!
My general feelings are:

1.  I want this package to stay focused on _automated_ time series forecasting. While I want lower level features to be usable if someone wants, I don't expect that to be the focus. Hand crafting pipelines makes data scientists feel good about themselves, but is often worse than (proper) automation.
2. This package is surprisingly complex. Lots of pieces interact with lots of other pieces. It is generally going to be much easier for me to add something in (by request) than to coach someone else through really understanding the code base. I have tried, and even talking about it for a couple of hours, leaves people still quite confused
3.  I enjoying being a not-so-[benevolent dictator for life](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life).

**Reply by winedarksea** [reply]
I used to use tsfresh in AutoTS back in the early days but it was way too slow. The closest I have is the much simpler: https://github.com/winedarksea/AutoTS/blob/1ef829b5eb706d076fdb650927fa5f1c5586bf7e/autots/models/sklearn.py#L38C11-L38C11
and the seasonal.py folder stuff

**Comment by rcyost** [comment]
Totally get all those points. 

even talking about it for a couple of hours, leaves people still quite confused <- record yourself and put it on youtube! after a bit you'll learn where your explanation was confused, and repeat until you can explain it as succinctly as it can be. at some point the opaqueness out ways the benefits of using this package... 

O the feature engineering problem... feature engineering is often more important than model selection. So this package is imcomplete without an automated feature engineering system as well. Fast or slow, it should still have this aspect of a complete auto time series pipeline. A potential  issue with the way that features are handled now is that scaling features ex post leaks information back instead of scaling at the time of the splits, right? create_regressor has the option to scale everything before the model is run, which leaks information about the future distribution of features back in time.
